soc/xtensa/esp32/Kconfig.soc

```diff
@@ -20,6 +20,22 @@ config ESPTOOLPY_FLASHFREQ_80M
 	bool
 	default y
 
+config FLASH_SIZE
+	int
+	default $(dt_node_reg_size_int,/soc/flash-controller@3ff42000/flash@0,0)
```

Comment:
@tejlmand I could not find a proper way to retrieve flash size from DTSI file. Also, default `FLASH_SIZE` option is missing for XTENSA and RISCV architectures (see https://gitlab.espressif.cn:6688/idf/zephyrproject-rtos/zephyr/-/blob/main/arch/Kconfig#L182)
Even if it was available, the default FLASH_SIZE in there does no return the DTSI flash controller reg size as needed in here. Any comments?

Reply:
what is wrong with the chosen `zephyr,flash` ?
https://github.com/zephyrproject-rtos/zephyr/blob/48c0faba1fb6e17ed4bec726caa3fac5de0bdb85/arch/Kconfig#L193-L197

Reply:
@tejlmand, 
1) FLASH_SIZE isn't available for xtensa and riscv as you can see in this condition: https://github.com/zephyrproject-rtos/zephyr/blob/48c0faba1fb6e17ed4bec726caa3fac5de0bdb85/arch/Kconfig#L190
2) Even if I add xtensa and riscv in there, CONFIG_FLASH_SIZE as defined in this Kconfig will return the value of this reg size: https://github.com/zephyrproject-rtos/zephyr/blob/48c0faba1fb6e17ed4bec726caa3fac5de0bdb85/dts/xtensa/espressif/esp32.dtsi#L69. However, it needs this reg value: https://github.com/zephyrproject-rtos/zephyr/blob/48c0faba1fb6e17ed4bec726caa3fac5de0bdb85/dts/xtensa/espressif/esp32.dtsi#L78

So, the approach I used is similar to what other vendors.  Do you have any recommendation?